walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,68 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> eventRanges = sortedRangesInQuestion(events, request.getAttendees());
+    List<TimeRange> possibleRanges = new ArrayList<TimeRange>();
+    int startingPointer = 0;
+    int endingPointer = 1;
+    System.out.println(eventRanges.size());
+    while(endingPointer < eventRanges.size()) {
+      if(eventRanges.get(startingPointer).contains(eventRanges.get(endingPointer))) {
+        endingPointer++;
+      }
+      else if(eventRanges.get(startingPointer).overlaps(eventRanges.get(endingPointer))) {
+        startingPointer = endingPointer;
+        endingPointer++;
+      }
+      else {
+        TimeRange between = TimeRange.fromStartEnd(eventRanges.get(startingPointer).end(),
+            eventRanges.get(endingPointer).start(), false);
+        if(between.duration() >= request.getDuration()){
+          possibleRanges.add(between);
+        }
+        startingPointer++;
+        endingPointer++;
+      }
+    }
+    if(eventRanges.size() >= 1){
+      TimeRange startOfDay = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, eventRanges.get(0).start(), false);
+      if(startOfDay.duration() >= request.getDuration()){
+        possibleRanges.add(startOfDay);
+      }
+      TimeRange latestEnd = eventRanges.get(0);
+      for(TimeRange time : eventRanges){
+        if(time.end() > latestEnd.end()){
+            latestEnd = time;
+        }
+      }
+      TimeRange endOfDay = TimeRange.fromStartEnd(latestEnd.end(), TimeRange.END_OF_DAY, true);
+      if(endOfDay.duration() >= request.getDuration()){
+        possibleRanges.add(endOfDay);
+      }
+    }
+    if(eventRanges.size() == 0 && request.getDuration() < TimeRange.WHOLE_DAY.duration()){
+      TimeRange fullDay = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true);
+      possibleRanges.add(fullDay);
+    }
+    Collections.sort(possibleRanges, TimeRange.ORDER_BY_START);
```

Comment:
I think you would only need this sort statement because you add the start of the day meetings at the very end of the process (so that possibleRanges has all the available meeting times in the middle of the day, then the start of the day meeting, then the end of the day meeting). 

Since it doesn't look like this code uses your start and end pointers or the results of the while loop, you could the startOfDay to before line 30 (so before the while loop) and leave the endOfDay calculation where it is, at the end, without needing to sort it again?

Let me know if this makes sense!